ControllableHero.hx

```diff
@@ -0,0 +1,49 @@
+package;
+
+    import flixel.FlxSprite;
+    import flixel.FlxG;
+
+    class ControllableHero extends FlxSprite {
+        
+        public function new(x:Int = 0, y:Int = 0){
+            super(x,y);
+            super.makeGraphic(15, 15);
```

Comment:
Numbers like these `15`s can be pretty mysterious, especially if we step away from the codebase for  a while. 

Are these both 15 because they were supposed to always match one another? Were they independently picked, and can be changed willy-nilly without issues? Are there any other spots in the code that depend on these staying unchanged?

It's good practice to take numbers like these and store them in class-level variables - static properties that can't be set to something new later is usually a good policy for const-type stuff.

Then we can give them names and make the intent of our code clearer - the player's starting_width should be 15, the player's starting_height should be 15.

Let's implement this sort of fix here and elsewhere in the project.

Reply:
Just pushed another commit after resolving all of your feedback. Thanks for all the optimization tips! Let me know if there are any other issues. I am specifically unsure of 
-whether my functions called in the update() should be declared as private or public, so I went with public
-location of my variable declarations. I think I put everything right at the top of the class declarations, outside and before any methods/functions

Reply:
Thanks for making the updates, I'm reviewing it all now.

About the public vs. private question:

It's useful to make functions public if you expect that other classes would have a reason to trigger that functionality. `update()` and `hurt()` are good examples of methods that the Haxeflixel developers decided to make public - 
 * We know we want our currently-running `FlxState` to be responsible for calling everyone's `update` method. Exposing `update()` as public a public method is what makes coordinating every object's tick-by-tick updates possible.
 * `hurt()` might be a bit tricky. We could organize our code in a way that makes our `FlxSprite`-inheriting classes always responsible for managing changes to their health. For example, Mario could always directly check for any interaction that hurts him, and damage himself accordingly when that happens.

   That plan sharply limits how our classes can interact with one another, though, and might lead to more complicated code overall - now Mario's always checking for some interaction with goombas, koopas, piranha plants, etc., and goombas et. al are always checking for some interaction with Mario.

   Exposing something like `hurt()` means that we get the choice of making something like the FlxState responsible for telling each involved party how to resolve some interaction. This can come with its own risks of tangled and tightly-coupled code, but we can see a reason that we'd want to let it be used by some external class.

For the methods you created like `keyboardMovement()` and `enforceBoundaries()`, we can use the same sort of logic to decide whether they're private or public. 

If there's a beneficial reason to expose them (there's a reason why other classes would be able to trigger this functionality) then we should make them public. If we don't want to expose these actions as something other classes can make our Screensaver / Controllable Hero do, then it's better to hide them away as private methods. Anything declared as public should be created with the intent of supporting external use, and other programmers (or ourselves, when we come back to this project months later) will be able to use its accessibility to get useful insight into what the method was intended for.

Reply:
The variable declarations look good to me. 

Defining data at the top is an organizational style that I find very readable. As your number of variables grow, it may be helpful to break them apart (separate them with an extra blank line) into whatever logical groupings make sense to you. The way these are set up is solid, though, especially because we don't have too many to worry about yet.